src/Manager.js

```diff
@@ -2,48 +2,50 @@
 import * as React from 'react';
 import createContext, { type Context } from 'create-react-context';
 
-export const ManagerContext: Context<{
+type ContextData = {
   setReferenceNode?: (?HTMLElement) => void,
   referenceNode?: ?HTMLElement,
-}> = createContext({ setReferenceNode: undefined, referenceNode: undefined });
+};
+
+export const ManagerContext: Context<ContextData> = createContext({ setReferenceNode: undefined, referenceNode: undefined });
 
 export type ManagerProps = {
   children: React.Node,
 };
-type ManagerState = {
-  context: {
-    setReferenceNode?: (?HTMLElement) => void,
-    referenceNode?: ?HTMLElement,
-  },
-};
 
 export default class Manager extends React.Component<
-  ManagerProps,
-  ManagerState
+  ManagerProps
 > {
+  contextObj: ContextData;
+
   constructor() {
     super();
-    this.state = {
-      context: {
+    this.contextObj = {
         setReferenceNode: this.setReferenceNode,
         referenceNode: undefined,
-      },
     };
   }
 
   setReferenceNode = (referenceNode: ?HTMLElement) => {
-    if (!referenceNode || this.state.context.referenceNode === referenceNode) {
+    if (!referenceNode || this.contextObj.referenceNode === referenceNode) {
       return;
     }
 
-    this.setState(({ context }) => ({
-      context: { ...context, referenceNode },
-    }));
```

Comment:
I had to remove the usage of `state` because React doesn't want `state` to be updated in `componentWillUnmount`. To get around this, I utilized class attributes to prevent new objects being created for the `value` property and `forceUpdate` to trigger a re-render when the `referenceNode` is updated.

From everything I can tell based on the git history, I _think_ this continues to support the use cases for this class, but happy to evaluate other approaches if I am missing something.

Reply:
Yes this is correct approach, storing html element in state was not nice. Good work 👍 

We can also split ManagerContext into two, one responsible for setting the element and another for passing the element... Then when Reference component will call ref it will no be re-rendered.

Something like this https://codesandbox.io/s/popper-context-lig79

Reply:
That's a good call - it negates the constraint of React always re-rendering consumers of the context for new objects.

Reply:
Updated